desktop/lib/reloadable.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Dev utility for server-side code reloading without the restart. In development,
+ * watch for file-system changes and clear cached modules when a change occurs,
+ * thus effectively reloading the entire app on request.
+ *
+ * @example
+ *
+ * // index.js
+ * import 'babel-core/register'
+ * import express from 'express'
+ * import reloadable, { isDevelopment } from 'desktop/lib/reloadable'
+ *
+ * const app = module.exports = express()
+ *
+ * app.set('view engine', 'jade')
+ * app.set('views', `${__dirname}/templates`)
+ *
+ * if (isDevelopment) {
+ *   reloadable(app, (req, res, next) => {
+ *     require('./server')(req, res, next)
+ *   }
+ * } else {
+ *   app.use(require('./server'))
+ * }
+ *
+ * // server.js
+ * import * as routes from './routes'
+ * import express from 'express'
+ *
+ * const app = module.exports = express.Router()
+ *
+ * app.get('/foo/:bar', routes.index)
+ * ...
+ */
+
+import path from 'path'
+import invariant from 'invariant'
+import { isFunction } from 'underscore'
+
+export const isDevelopment = process.env.NODE_ENV === 'development'
+
+export default function reloadable (app, onReload, appPath) {
+  if (isDevelopment) {
```

Comment:
Can you explain a bit more? We have to guard against different envs, and I'd want to throw if the user is trying to mount this util outside of the guard.